Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyEntityIdCtrl.cpp

```diff
@@ -331,6 +335,23 @@ namespace AzToolsFramework
 
     void PropertyEntityIdCtrl::SetCurrentEntityId(const AZ::EntityId& newEntityId, bool emitChange, const AZStd::string& nameOverride)
     {
+        if (GetCheckClearButton() && newEntityId.IsValid())
+        {
+            // First check if the new entity has a parent.
+            AZ::EntityId parentId;
+            EditorEntityInfoRequestBus::EventResult(parentId, newEntityId, &EditorEntityInfoRequestBus::Events::GetParent);
+            if (!parentId.IsValid())
+            {
+                // If not then disable the clear button.
+                m_entityIdLineEdit->setClearButtonEnabled(false);
+            }
+            else
+            {
+                bool isPrefabEntity = (m_prefabPublicInterface) ? m_prefabPublicInterface->IsInstanceContainerEntity(newEntityId) : false;
+                // if the parent is a prefab container then disable the clear button else enable it.
+                m_entityIdLineEdit->setClearButtonEnabled(isPrefabEntity ? false : true);
+            }
+        }
```

Comment:
I don't think the UI classes (PropertyEntityIdCtrl) should implement custom logic based on prefabs behavior.
I think this check should actually be using the Editor Focus Mode interface to verify whether the entity is the focus root, and if so, hide the clear button. This way we lose the dependency on prefabs (editor focus mode is an editor-side concept, prefabs only happen to use it)

Reply:
Yeah, the issue being resolved (#7465) is only meant to hide the clear button for the Parent EntityId field on a Transform component. In this current form, this logic would be getting applied to all EntityId fields in any property editor.

Reply:
Not really, since GetCheckClearButton checks an attribute that is only set for the parent property of the transform.
I still find it confusing though, the attribute naming is not clear, and if possible I'd rather have the ability to set a validation function via EditContext and then have that be defined in the Transform component, with no logic on the UI side other than triggering it and getting the result out.

Reply:
No problem changing to the EditContext, but not sure what you mean. Is there an example somewhere?